x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptFieldMapper.java

```diff
@@ -56,81 +73,118 @@ protected String contentType() {
         return CONTENT_TYPE;
     }
 
-    public static class Builder extends FieldMapper.Builder<Builder> {
-
+    public static class Builder extends ParametrizedFieldMapper.Builder {
+
+        private final Parameter<Map<String, String>> meta = new Parameter<>(
+            "meta",
+            true,
+            Collections.emptyMap(),
+            TypeParsers::parseMeta,
+            m -> m.fieldType().meta()
+        );
+        private final Parameter<String> runtimeType = Parameter.stringParam(
+            "runtime_type",
+            true,
+            mapper -> toType(mapper).runtimeType,
+            null
+        );
+        // TODO script and runtime_type can be updated: what happens to the currently running queries when they get updated?
+        // do all the shards get a consistent view?
```

Comment:
@romseygeek do you have thoughts on this?

Reply:
So currently `QueryShardContext#fieldMapper()` delegates field lookups back to the MapperService, which gets updated in-place, so if there's an update while a query is running then the script will get changed out and there may well be inconsistencies between phases, as well as between shards.  We could make QueryShardContext take a FieldTypeLookup as a parameter which would at least remove the possibility of an update between phases?